process_metrics_linux.go

```diff
@@ -41,13 +42,35 @@ type procStat struct {
 	Rss         int
 }
 
+var processSelfIONotFoundErrorCount, processSelfIOPermErrorCount, processSelfErrorCount int64
+
+func init() {
+	ioFilepath := "/proc/self/io"
+	_, err := ioutil.ReadFile(ioFilepath)
+	if err != nil {
+		// Do not spam the logs with errors
+		// This error will not be fixed without process restart
+		var errMsg string
+		switch {
+		case os.IsNotExist(err):
+			errMsg = fmt.Sprintf("ERROR: metrics: cannot open %q: %s. This is expected on kernel without CONFIG_TASK_IO_ACCOUNTING, systems without cgroup controller for IO. This error will be reported once, further errors can be tracked by 'process_io_stats_read_errors_total{reason=\"not_found\"}' metric", ioFilepath, err)
+		case os.IsPermission(err):
+			errMsg = fmt.Sprintf("ERROR: metrics: cannot open %q: %s. This is expected when process is running with limited permissions and capabilities (such as using systemd limitations, cgroups, selinux, apparmor and others). This error will be reported once, further errors can be tracked by 'process_io_stats_read_errors_total{reason=\"permission_denied\"' metric", ioFilepath, err)
+		default:
+			errMsg = fmt.Sprintf("ERROR: metrics: cannot open %s: %s", ioFilepath, err)
+		}
+		log.Print(errMsg)
+	}
```

Comment:
let's be well-mannered and close the file)

Reply:
`ioutil.ReadFile` function returns `[]byte, err` but not the file, so we don't need to close it here